src/OpenLoco/src/GameCommands/VehicleChangeRunningMode.cpp

```diff
@@ -0,0 +1,173 @@
+        if (startVehicle && !train.head->hasVehicleFlags(VehicleFlags::commandStop))
+        {
+            return 0;
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        train.head->vehicleFlags ^= VehicleFlags::commandStop;
+        if (train.head->hasVehicleFlags(VehicleFlags::commandStop))
+        {
+            train.head->vehicleFlags &= ~VehicleFlags::manualControl;
+        }
+
+        if (!(train.head->hasVehicleFlags(VehicleFlags::unk_2) || !CompanyManager::isPlayerCompany(train.head->owner)))
+        {
+            auto madeProfit = train.veh2->profit[0] | train.veh2->profit[1] | train.veh2->profit[2] | train.veh2->profit[3];
+            if (madeProfit != 0)
+            {
+                StationManager::sub_437F29(getUpdatingCompanyId(), 1);
+            }
+        }
+
+        invalidateWindow(train.head->head);
+        return 0;
+    }
+
+    // 0x004B6AAF
+    static uint32_t toggleManualDriving(const Vehicles::Vehicle& train, const uint8_t flags)
+    {
+        // Can we change driving modes?
+        if (!sub_4B6B0C(train.head))
+        {
+            return FAILURE;
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        train.head->vehicleFlags ^= VehicleFlags::manualControl;
+        train.head->var_6E = -40;
+
+        if ((train.head->vehicleFlags & VehicleFlags::manualControl) != VehicleFlags::none)
+        {
+            train.head->vehicleFlags &= ~VehicleFlags::commandStop;
+        }
+        else
+        {
+            train.head->vehicleFlags |= VehicleFlags::commandStop;
+        }
+
+        if (train.head->status == Vehicles::Status::approaching)
+        {
+            train.head->stationId = StationId::null;
+            train.head->status = Vehicles::Status::travelling;
+        }
+
+        invalidateWindow(train.head->head);
+
+        return 0;
+    }
+
+    // 0x004B694B
+    static uint32_t vehicleChangeRunningMode(const VehicleChangeRunningModeArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
+        {
+            Vehicles::Vehicle train(args.head);
+            setPosition(train.veh2->position);
+
+            // If we're only changing local/express, we can take a shortcut.
+            if (args.mode == VehicleChangeRunningModeArgs::Mode::toggleLocalExpress)
+            {
+                return changeLocalExpressMode(train, flags);
+            }
+
+            // If a vehicle is stuck or crashed, immediately sell the vehicle instead.
+            if ((train.head->status == Vehicles::Status::stuck || train.head->status == Vehicles::Status::crashed) && CompanyManager::isPlayerCompany(train.head->owner))
+            {
+                if (flags & Flags::apply)
+                {
+                    train.head->sub_4AD778();
+                    train.head->status = Vehicles::Status::stopped;
```

Comment:
there is also something setting `train.veh2->currentSpeed = 0`

Reply:
Thanks, I'd really looked over that.